packages/core/src/service/services/order.service.ts

```diff
@@ -441,6 +448,27 @@ export class OrderService {
         this.eventBus.publish(new OrderStateTransitionEvent(fromState, state, ctx, order));
         return order;
     }
+    async findOrderByFulfillmentId(fulfillment: Fulfillment): Promise<Order> {
+        const order = await this.connection.getRepository(Order).findOneOrFail({
```

Comment:
Orders are now Channel-aware, so you should also add a channel check to the `where` clause here.